src/fe/inf_fe_map.C

```diff
@@ -29,6 +29,18 @@
 namespace libMesh
 {
 
+bool libmesh_isfinite(Real value)
+{
+  // check if \p value is infinite
+  if( value > std::numeric_limits<Real>::max()
+         || value <  std::numeric_limits<Real>::min() )
```

Comment:
Maybe you are right that the comparison is not very smart; and the suggestion in your link seems to be much more elegant to me.
But if you are working on this, I am fine. Thank you!